syncback/Tiltfile

```diff
@@ -20,25 +22,56 @@ def syncback(name, k8s_object, src_dir, paths=None, target_dir='.', container=''
            object, using the first container by default.
     :param src_dir (str): directory IN THE KUBERNETES CONTAINER to sync from. Any paths specified, if relative,
            should be relative to this dir.
+    :param ignore (List[str], optional): files to ignore when syncing (relative to src_dir).
+    :param delete (bool, optional): run rsync with the --delete flag, i.e. delete files locally if not present in
+           the container. By default, False. THIS OPTION RISKS WIPING OUT FILES that exist locally but not in the
+           container. Tilt will protect some files automatically, but we recommend syncing specific paths (via `paths`
+           and/or using the `ignore` parameter to explicitly protect other files that exist locally but not on the container.
     :param paths (List[str], optional): paths IN THE KUBERNETES CONTAINER to sync, relative to src_dir. May be files or dirs.
-           Note that these must not begin with `./`. If no paths are provided, sync the entire src_dir.
+           Note that these must not begin with `./`. If this arg is not passed, sync all of src_dir.
     :param target_dir (str, optional): directory ON THE LOCAL FS to sync to. Defaults to '.'
     :param container (str, optiona): name of the container to sync from (by default, the first container)
     :param namespace (str, optiona): namespace of the desired k8s_object, if not `default`.
     :param verbose (bool, optional): if true, print additional rsync information.
     :return:
     """
 
+    # Verify inputs
+    if not src_dir.endswith('/'):
+        fail('src_dir must be a directory and have a trailing slash (because of rsync syntax rules)')
+
+    if paths:
+        for p in paths:
+            if p.startswith('./'):
+                fail('Found illegal path "{}": paths may not begin with ./ (because of rsync syntax rules)'.format(p))
+            if p.startswith('/'):
+                fail('Found illegal path "{}": paths may not begin with / and must be relative to src_dir (because of rsync syntax rules)'.format(p))
+
+    # Construct include/exclude rules
     incl_excl = ''
+
+    to_exclude = ignore
+    if not ignore:
+        to_exclude = []
+
     if paths:
         # TODO: if you're rsync-savvy you might want to do the wildcarding manually--
         #   give an option to turn off automatic +'***'
         includes = ' '.join(['--include="{}***"'.format(p) for p in paths])
         incl_excl = '{} --exclude="*"'.format(includes)
+    else:
+        # Sync the entire src_dir. Danger, Will Robinson! Exclude some stuff
+        # that may exist locally but not in your container so it
+        # doesn't get wiped out locally on your first sync
+        to_exclude.extend(DEFAULT_EXCLUDES)
```

Comment:
i think this still has the same problem - it mutates an argument. probably simpler to reason about if you simply don't use mutation (e.g., do to_exclude = DEFAULT_EXCLUDES + to_exclude)